src/extractor/restriction_compressor.cpp

```diff
@@ -0,0 +1,104 @@
+#include "extractor/restriction_compressor.hpp"
+#include "extractor/restriction.hpp"
+
+#include <algorithm>
+#include <boost/assert.hpp>
+#include <utility>
+
+namespace osrm
+{
+namespace extractor
+{
+
+RestrictionCompressor::RestrictionCompressor(std::vector<TurnRestriction> &restrictions)
+{
+    // add a node restriction ptr to the heads/tails maps, needs to be a reference!
+    auto index = [&](auto &element) {
+        heads.insert(std::make_pair(element.from, &element));
```

Comment:
Terminology wise, the `head` is typically the target of an edge, here it is the `from` of the restriction. Maybe `restriction_starts` and `restrictions_targets` would be more obvious?